pkg/services/alerting/notifiers/wecom.go

```diff
@@ -0,0 +1,237 @@
+package notifiers
+
+import (
+	"crypto/md5"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/grafana/grafana/pkg/bus"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/alerting"
+)
+
+const weComMsgTypeNews = "news"
+const weComMsgTypeMarkdown = "markdown"
+
+func init() {
+	alerting.RegisterNotifier(&alerting.NotifierPlugin{
+		Type:        "wecom",
+		Name:        "WeCom",
+		Description: "Sends HTTP POST request to WeCom",
+		Heading:     "WeCom settings",
+		Factory:     NewWeComNotifier,
+		Options: []alerting.NotifierOption{
+			{
+				Label:        "Url",
+				Element:      alerting.ElementTypeInput,
+				InputType:    alerting.InputTypeText,
+				Placeholder:  "https://qyapi.weixin.qq.com/cgi-bin/webhook/send?key=xxxxxxxx",
+				PropertyName: "url",
+				Required:     true,
+			},
+			{
+				Label:        "Message type",
+				Element:      alerting.ElementTypeSelect,
+				PropertyName: "msgType",
+				SelectOptions: []alerting.SelectOption{
+					{
+						Value: weComMsgTypeMarkdown,
+						Label: "Markdown",
+					},
+					{
+						Value: weComMsgTypeNews,
+						Label: "News",
+					},
+				},
+			},
+			{
+				Label:        "Send image as single message",
+				Element:      alerting.ElementTypeCheckbox,
+				InputType:    alerting.ElementTypeCheckbox,
+				Description:  "Send included image as individual message for a clearer reading",
+				PropertyName: "uploadSingleImage",
+			},
+		},
+	})
+}
+
+// NewWeComNotifier is the constructor for WeCom notifier.
+func NewWeComNotifier(model *models.AlertNotification, _ alerting.GetDecryptedValueFn) (alerting.Notifier, error) {
+	url := model.Settings.Get("url").MustString()
+	if url == "" {
+		return nil, alerting.ValidationError{Reason: "Could not find url property in settings"}
+	}
+
+	msgType := model.Settings.Get("msgType").MustString(weComMsgTypeMarkdown)
```

Comment:
At this point, we need to validate that the message type is either `weComMsgTypeNews` or `weComMsgTypeMarkdown`.

I recommend doing the following:

```golang
var allowedWeComMsgTypes := map[string]struct{}{
  weComMsgTypeNews: {},
  weComMsgTypeMarkdown: {},
}

// Then you can simply do
_, allowed := allowedWeComMsgTypes[msgType]
if !allowed {
  return nil, fmt.Errorf("")
}
```